modin/engines/base/frame/data.py

```diff
@@ -1719,53 +1728,36 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         if isinstance(other, type(self)):
             other = [other]
 
-        is_aligning_applied = False
-        for i in range(len(other)):
-            if (
-                len(self._partitions) != len(other[i]._partitions)
-                and len(self.axes[0]) == len(other[i].axes[0])
-                and axis == 0
-            ):
-                is_aligning_applied = True
-                self._partitions = self._frame_mgr_cls.map_axis_partitions(
-                    axis, self._partitions, lambda df: df
-                )
-                other[i]._partitions = other[i]._frame_mgr_cls.map_axis_partitions(
-                    axis, other[i]._partitions, lambda df: df
-                )
-
-        if (
-            all(o.axes[axis].equals(self.axes[axis]) for o in other)
-            and not is_aligning_applied
-        ):
+        if all(o.axes[axis].equals(self.axes[axis]) for o in other):
             return (
                 self._partitions,
                 [self._simple_shuffle(axis, o) for o in other],
-                self.axes[axis].copy(),
             )
 
-        index_other_obj = [o.axes[axis] for o in other]
-        joined_index = self._join_index_objects(axis, index_other_obj, how, sort)
         # We have to set these because otherwise when we perform the functions it may
         # end up serializing this entire object.
         left_old_idx = self.axes[axis]
-        right_old_idxes = index_other_obj
+        right_old_idxes = [o.axes[axis] for o in other]
 
-        def make_map_func():
-            if not joined_index.is_unique and axis == 0:
-                return lambda df: df
-            return lambda df: df.reindex(joined_index, axis=axis)
+        if make_map_reindexer is None:
+
+            def make_map_func(index, *args, **kwargs):
+                # left - specific argument for case of binary operation;
```

Comment:
Is this comment make sense this? `left` parameter isn't presented here.

Reply:
Removed.